modules/m_trace.c

```diff
@@ -381,13 +383,22 @@ report_this_status(struct Client *source_p, struct Client *target_p)
 
 	case STAT_CLIENT:
 		{
+			/* fire the doing_trace_show_idle hook to allow modules to tell us whether to show the idle time */
+	                hook_data_client_approval hdata_showidle;
+
+	                hdata_showidle.client = source_p;
+	                hdata_showidle.target = target_p;
+	                hdata_showidle.approved = 1;
+
+			call_hook(doing_trace_show_idle_hook, &hdata_showidle);
+
 			sendto_one_numeric(source_p,
 					SeesOper(target_p, source_p) ? RPL_TRACEOPERATOR : RPL_TRACEUSER,
 					SeesOper(target_p, source_p) ? form_str(RPL_TRACEOPERATOR) : form_str(RPL_TRACEUSER),
 					class_name, name,
 					show_ip(source_p, target_p) ? ip : empty_sockhost,
-					(unsigned long)(rb_current_time() - target_p->localClient->lasttime),
-					(unsigned long)(rb_current_time() - target_p->localClient->last));
+					hdata_showidle.approved ? (unsigned long)(rb_current_time() - target_p->localClient->lasttime) : 0,
```

Comment:
`stats l` uses -1 to show it's hiding the value, but this uses 0 still

Reply:
IRC discussion in a nutshell:
```
11:43:30 < launchd> does anybody have opinions on https://github.com/solanum-ircd/solanum/pull/220#discussion_r720620067
11:59:02 < dwfreed> hexley: I vote 0
11:59:33 < dwfreed> whatever's reading this may not be expecting a negative number
11:59:53 < dwfreed> and moreso may not be expecting a non-number
12:49:10 < spb> if there's another part of the protocol that already uses 0 for 'idle time not known', then i'm not sure how much of a debate there is to be had about what to use in the new case
```